libs/datamodel/core/src/transform/ast_to_dml/lift.rs

```diff
@@ -479,12 +483,11 @@ fn datamodel_connector_native_type_to_dml_native_type(
     }
 }
 
-fn dml_default_kind(default_value: DefaultValueWalker<'_, '_>) -> dml::DefaultKind {
+fn dml_default_kind(default_value: &ast::Expression, scalar_type: Option<ScalarType>) -> dml::DefaultKind {
```

Comment:
nitpick: was this refactoring needed? I see it's about the same complexity, but in a different way...

Reply:
Yes, since `DefaultValueWorker` expects an`ast::FieldId`, making it work only with model fields. The point of this refactor is to allow sharing the conversion between model & composites.